components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAO.java

```diff
@@ -48,6 +48,9 @@ void deactivateAuthorizationCode(AuthzCodeDO authzCodeDO) throws
 
     Set<String> getAuthorizationCodesByUser(AuthenticatedUser authenticatedUser) throws IdentityOAuth2Exception;
 
+    List<AuthzCodeDO> getAuthorizationCodesDataByUser(AuthenticatedUser authenticatedUser)
```

Comment:
Shouldn't we make this default because if someone else using the same interface, they might need to do a migration after this change